runtime/src/bank.rs

```diff
@@ -957,6 +958,13 @@ pub(crate) struct BankFieldsToSerialize<'a> {
     pub(crate) stakes: &'a StakesCache,
     pub(crate) epoch_stakes: &'a HashMap<Epoch, EpochStakes>,
     pub(crate) is_delta: bool,
+    /// The list of roots which are no longer roots in the current bank instance.
+    /// As we eliminate rewrites/ancient append vec, we need to the ability to
+    /// remember slots within that last epoch that WERE roots even if they no
+    /// longer contain any accounts.  Without remembering those prior roots,
+    /// accounts that we skip rewrites might have their rent collection time
+    /// point to the incorrect roots as their correct roots were removed.
+    pub(crate) prior_roots: Vec<Slot>,
```

Comment:
@carllin is this a sufficient data structure to store what you were looking into regarding slashing and such?

Reply:
I think it'll require `(Slot, Hash)` to be perfectly safe, but I wouldn't worry about that for now if that imposes a big burden.

Reply:
Hash = bank hash at that slot?
Are you confident this is useful info for what you're doing? Do we keep around the bank hashes for past slots during the entire epoch? I can look. Just thought you may know.
I could just store default hashes at the moment or Option<Hash> with None. So, they'd be present in the format, but not populated.